### PR TITLE
chore(package): update all dependencies

### PR DESCRIPTION
**Pull request by layershifter**
# WIP

This PR:
- updates all deps to their latest version
- removes usage of `eslint-config-ta`, vanilla `airbnb` config will be used
- removes `del`, we have already `rimraf` for such cases
- removes `empty`, cruft

# TODO

- [x] fix lint errors
- [x] ES-ify configs
- [x] clean up `eslint` configs
- [x] merge #1938
- [x] merge #1939
- [x] merge #1949
- [x] replace [`babel-plugin-react-transform`](https://github.com/gaearon/babel-plugin-react-transform), it's deprecated (#1946)
- [x] replace [`babel-plugin-__coverage__`](https://github.com/dtinth/babel-plugin-__coverage__), it's deprecated (#1940)

### CODE REVIEW

**Comment by levithomason** [comment]
While we're at this, I've been testing out [`prettier`](https://github.com/prettier/prettier) in other projects and making some suggestions over there as well.  I'm curious what our codebase looks like with it.  I know it would be a drastic change in many ways, but it is one less thing to have to manage.

**Comment by layershifter** [comment]
@levithomason I've splitted only style changes to separate PRs (#1938, #1939, #1949), after they will be merged I'll continue work. Please merge them, because they will cause merge conflicts.

I investigated situation with `karma-webpack-with-fast-source-maps` and there are no alternatives to it. In fact `karma-webpack` should do same, but it doesn't. However, `karma-webpack-with-fast-source-maps` doesn't support `webpack@3` and its author doesn't plan to add it. I will check this in separate PR, possible I can simply update `peerDependency` in its `package.json`.

Also, `sinon@3` was released, but packages that dependens on it aren't updated yet.

**Comment by levithomason** [comment]
All blocking PRs are merged.

**Comment by layershifter** [comment]
Good news, `karma-webpack-with-fast-source-maps` was updated to support `webpack@3` and `sinon`'s deps also support `3.x` :+1: 

I need to cleanup `eslint` configs and review changes.

File: webpack.config.babel.js [hunk 1]
```diff
@@ -1,9 +1,11 @@
...
-const config = require('./config')
+import HtmlWebpackPlugin from 'html-webpack-plugin'
+import _ from 'lodash'
+import webpack from 'webpack'
+
+import config from './config'
+
 const { paths } = config
-const { __DEV__, __STAGING__, __TEST__, __PROD__ } = config.compiler_globals
+const { __DEV__, __TEST__, __PROD__ } = config.compiler_globals
```
Comment (layershifter) [inline webpack.config.babel.js:8]: `__STAGING__` is cruft

File: test/specs/modules/Checkbox/Checkbox-test.js [hunk 2]
```diff
@@ -59,8 +58,7 @@ describe('Checkbox', () => {
   describe('indeterminate', () => {
     it('can be indeterminate', () => {
       const wrapper = mount(<Checkbox indeterminate />)
-
-      const checkboxNode = findDOMNode(wrapper.instance())
+      const checkboxNode = wrapper.getDOMNode()
```
Comment (layershifter) [inline test/specs/modules/Checkbox/Checkbox-test.js:61]: Enzyme has a `getDOMNode`, so `findDOMNode` is useless there

**Review by layershifter — COMMENTED** [review]
We're are ready for review there

**Comment by levithomason** [comment]
This is ready to merge, we'll have quite a few conflicts on other branches I'm sure 😕 Onward!

**Comment by levithomason** [comment]
Released in `semantic-ui-react@0.71.5`

**Pull request merged**